test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala

```diff
@@ -356,6 +356,34 @@ class ExportsTest {
     assertEquals(4, bar.method(2))
   }
 
+  @Test def should_inherit_exports_from_traits_with_value_classes(): Unit = {
```

Comment:
This test already passes before the changes in this commit. It's only related because it also relies on the type history of export bridges.

Reply:
Well, turns out it *doesn't* pass in 2.10. It seems to be an independent issue, though, so I filed it as #3585.